arxiv/metadata/tests/test_metacheck.py

```diff
@@ -0,0 +1,432 @@
+# Tests for internal (helper) function combine_dispositions
+
+def test_combine_dispositions():
+    assert HOLD == combine_dispositions( HOLD, HOLD )
+    assert HOLD == combine_dispositions( HOLD, WARN )
+    assert HOLD == combine_dispositions( WARN, HOLD )
+    assert HOLD == combine_dispositions( HOLD, OK )
+    assert HOLD == combine_dispositions( OK, HOLD )        
+    assert WARN == combine_dispositions( WARN, WARN )
+    assert WARN == combine_dispositions( WARN, OK )
+    assert WARN == combine_dispositions( OK, WARN )
+    assert OK == combine_dispositions( OK, OK )
+
+############################################################l    
+##### TITLE field checks
+
+TITLE_TESTS = [
+    ('A fine title', None),
+    ('Another title about CERN and ALPEH where z~1/2', None),
+    ('',
+     (HOLD, ["Title cannot be empty"])),
+    ('Tiny',
+     (WARN, ["Title: too short"])),
+    ('a title with lowercase',
+     (WARN, ["Title: starts with a lower case letter"])),
+    ('aTITLE: Not So Lowercase', None),
+    # NOTE: we don't check for extra spaces
+    # ['  A title with leading space',"Title: leading spaces"],
+    # ['A title with trailing space ',"Title: trailing spaces"],
+    # ['A title  with  mutliple  spaces',"Title: multiple consecutive spaces"],
+    # NOTE: Titles can end with punctuation.
+    ('A title with period.', None),
+    # NOTE: the test for excessive capitalization is fragile
+    ('ALL CAPS TITLE',
+     (WARN, ["Title: excessive capitalization"])),
+    ('NOT EVEN BORDERLINE ALL CAPS TITLE',
+     (WARN, ["Title: excessive capitalization"])),
+    ('BORDERLINE All Caps TITLE', None),
+    ('BORDERLINE ALL caps TITLE', 
+     (WARN, ["Title: excessive capitalization"])),
+    ('This is a title WITH ONE LONG WORD CAPITALIZED', None),
+    ('This is a title WITH SOME EXTRAEXTRA LONG WORDS CAPITALIZED',
+     [WARN, ["Title: excessive capitalization"]]),
+    # We allow some capitalized words
+    ('The is a title with known long words capitalized AMANDA CHANDRA', None),
+    # Check for *some* HTML entities
+    ('These should not be flagged as HTML: <x> <xyz> <ijk> <i> <b>', None),
+    ('Factor Ratio to Q<sup>2</sup> = 8.5 GeV<sup>2</sup>',
+     (WARN, ['Title: contains HTML'])),
+    ('A title with HTML<br/>linebreaks<br />there',
+     (WARN, ['Title: contains HTML'])),
+    ('Title: Something',
+     (WARN, ["Title: begins with 'title'"])),
+    ('This \\\\ is a line break',
+     (WARN, ["Title: contains TeX line break"])),
+    ('This \\ is not a line break', None),
+    ('Don\'t use \\href{...}, \\url{...}, \\emph, \\textbf, \\texttt, \\%, or \\#: Something',
+     (WARN, [
+         'Title: contains TeX \\href',
+         'Title: contains TeX \\url',
+         'Title: contains \\emph',
+         'Title: contains \\textbf',
+         'Title: contains \\texttt',
+         'Title: contains unnecessary escape: \\#',
+         'Title: contains unnecessary escape: \\%',
+     ])),
+]
+
+def test_titles():

```

Comment:
Consider parameterizeing this: https://docs.pytest.org/en/stable/how-to/parametrize.html
Once you learn how to do it it is quick, useful and has good debugging output.